root/static/scripts/common/utility/isDateEmpty.js

```diff
@@ -7,6 +7,17 @@
  * later version: http://www.gnu.org/licenses/gpl-2.0.txt
  */
 
+export function isDateObservableEmpty(
+  date: {
+    day: KnockoutObservable<number | null>,
+    month: KnockoutObservable<number | null>,
+    year: KnockoutObservable<number | null>,
```

Comment:
(These types might need changing too)
```suggestion
    year: KnockoutObservable<StrOrNum | null>,
```

Reply:
Seems only string actually (or null)